extensions/ql-vscode/src/queries-panel/queries-module.ts

```diff
@@ -0,0 +1,30 @@
+import { App, AppMode } from "../common/app";
+import { isCanary, showQueriesPanel } from "../config";
+import { DisposableObject } from "../pure/disposable-object";
+import { QueriesPanel } from "./queries-panel";
+
+export class QueriesModule extends DisposableObject {
+  private queriesPanel: QueriesPanel | undefined;
+
+  private constructor(readonly app: App) {
+    super();
+  }
+
+  private async initialize(app: App): Promise<void> {
```

Comment:
I think this and the static `initialize` func don't need to be async/return Promise.